components/Repo/Add.js

```diff
@@ -55,45 +82,146 @@ const styles = {
   })
 }
 
+const TemplatePicker = compose(
+  withT,
+  withRouter,
+  graphql(getTemplateRepos, {
+    options: () => ({
+      fetchPolicy: 'network-only'
+    }),
+    skip: ({ isTemplate }) => isTemplate
+  })
+)(({ t, data, schema, onChange, isTemplate }) => {
+  const [schemaOptions] = useState(
+    schemaKeys.map(key => ({
+      value: key,
+      text: t(`repo/add/template/${key}`, null, key)
+    }))
+  )
+  const [templateFilter, setTemplateFilter] = useState('')
+  const [template, setTemplate] = useState({
+    value: schema,
+    text: t(`repo/add/template/${schema}`, null, schema)
+  })
+
+  const templateOptions = useMemo(() => {
+    return schemaOptions
+      .concat(
+        (data?.reposSearch?.nodes || []).map(repo => ({
+          value: repo.latestCommit.document.meta.template,
+          text:
+            repo.latestCommit.document.meta.title ||
+            repo.id.split('/')[1].replace('-', ' '),
+          repoId: repo.id,
+          slug: repo.latestCommit.document.meta.slug || repo.id.split('/')[1]
+        }))
+      )
+      .filter(
+        ({ text }) =>
+          !templateFilter ||
+          text.toLowerCase().includes(templateFilter.toLowerCase())
+      )
+  }, [data, templateFilter])
+
+  return isTemplate ? (
+    <Dropdown
+      label='Vorlage'
+      items={schemaOptions}
+      value={schema}
+      onChange={item => {
+        onChange({
+          schema: item.value,
+          templateRepoId: undefined,
+          templatePrefix: undefined
+        })
+      }}
+    />
+  ) : (
+    <Loader
+      error={data.error}
+      loading={data.loading}
+      render={() => (
+        <Autocomplete
+          label='Vorlage'
+          value={template}
+          filter={templateFilter}
+          items={templateOptions}
+          onChange={newTemplate => {
+            setTemplate(newTemplate)
+            setTemplateFilter('')
+            onChange({
+              schema: newTemplate.value,
+              templateRepoId: newTemplate.repoId,
+              templatePrefix: newTemplate.slug
+                ? newTemplate.slug.replace('template-', '') + '-'
+                : undefined
+            })
+          }}
+          onFilterChange={newFilter => {
+            if (!template || template.text !== newFilter) {
+              setTemplateFilter(newFilter)
+            }
+          }}
+          icon={<SearchIcon size={30} style={{ color: colors.lightText }} />}
+        />
+      )}
+    />
+  )
+})
+
 class RepoAdd extends Component {
   constructor(props) {
     super(props)
+    const schema = schemaKeys.includes('article') ? 'article' : schemaKeys[0]
     this.state = {
+      schema,
       title: '',
-      template: templateKeys.includes('article') ? 'article' : templateKeys[0]
+      maxSlugLength: props.isTemplate ? 30 : 100
```

Comment:
The problem with doing this here is that this only runs when the component is remounted. When toggling between templates and documents this does not happen. So it will just use the limit of the initially loaded page. Generally always avoid state when possible. In a hooks world this would be an ideal candidate for a `useMemo`, here I guess an function that is called from `onSubmit` and `handleTitle`. Or make `handleTitle` return an error and always call it in `onSubmit`.